projects/plugins/jetpack/modules/masterbar/wp-posts-list/class-posts-list-page-notification.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Posts_List_Page_Notification file.
+ * Disable edit_post and delete_post capabilities for Posts Pages in WP-Admin and display a notification icon.
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+/**
+ * Class Posts_List_Page_Notification
+ *
+ * @package Automattic\Jetpack\Dashboard_Customizations
+ */
+class Posts_List_Page_Notification {
+
+	/**
+	 * Site's Posts page id
+	 *
+	 * @var int|null
+	 */
+	private $posts_page_id;
+
+	/**
+	 * If the Post_list contains the site's Posts Page
+	 *
+	 * @var bool
+	 */
+	private $is_page_in_list = false;
+
+	/**
+	 * Class instance.
+	 *
+	 * @var Posts_List_Page_Notification|null
+	 */
+	private static $instance = null;
+
+	/**
+	 * Posts_List_Page_Notification constructor.
+	 *
+	 * @param string $posts_page_id The Posts page configured in WordPress.
+	 */
+	public function __construct( $posts_page_id ) {
+		\add_filter( 'map_meta_cap', array( $this, 'disable_posts_page' ), 10, 4 );
+		\add_filter( 'post_class', array( $this, 'add_posts_page_css_class' ), 10, 3 );
+		\add_action( 'admin_print_footer_scripts-edit.php', array( $this, 'add_notification_icon' ) );
+
+		$this->posts_page_id = '' === $posts_page_id ? null : (int) $posts_page_id;
+	}
+
+	/**
+	 * Creates instance.
+	 *
+	 * @return Posts_List_Page_Notification
+	 */
+	public static function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self( \get_option( 'page_for_posts' ) );
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Disable editing and deleting for the page that is configured as a Posts Page.
+	 *
+	 * @param array  $caps Array of capabilities.
+	 * @param string $cap The current capability.
+	 * @param string $user_id The user id.
+	 * @param array  $args Argument array.
+	 * @return array
+	 */
+	public function disable_posts_page( $caps, $cap, $user_id, $args ) {
+		if ( 'edit_post' !== $cap && 'delete_post' !== $cap ) {
+			return $caps;
+		}
+
+		if ( isset( $args[0] ) && $this->posts_page_id === $args[0] ) {
```

Comment:
Are we sure that the `$args[0]` will always be what we expect for?

Reply:
It should be pretty safe I think, WPCore does it the same for delete_page and edit_page:
https://core.trac.wordpress.org/browser/trunk/src/wp-includes/capabilities.php#L76
https://core.trac.wordpress.org/browser/trunk/src/wp-includes/capabilities.php#L149